.ci/schedule-tests/src/combine-statistics.js

```diff
@@ -0,0 +1,53 @@
+const fs = require('fs')
+const path = require('path')
+const AdmZip = require('adm-zip')
+
+const processTextFile = (fileContent, durations) => {
+  const lines = fileContent.split(/\n/)
+
+  for (const line of lines) {
+    const [suite, testFile, duration] = line.split(/\s+/)
+    const durationInSeconds = parseFloat(duration)
+
+    if (!isNaN(durationInSeconds)) {
+      const key = `${suite}:${testFile}`
+      durations[key] = [...(durations[key] || []), durationInSeconds]
+    }
+  }
+}
+
+const calculateMedian = (arr) => {
+  const sortedArr = arr.sort((a, b) => a - b)
+  const middle = Math.floor(sortedArr.length / 2)
+
+  return sortedArr.length % 2 === 0
+    ? (sortedArr[middle - 1] + sortedArr[middle]) / 2
+    : sortedArr[middle]
+}
+
+const combineStatistics = (directoryPath, outputFilePath) => {
+  const durations = {}
+
+  const files = fs.readdirSync(directoryPath)
+  for (const file of files) {
+    if (file.endsWith('.zip')) {
+      const filePath = path.join(directoryPath, file)
```

Comment:
I'm not using `forEach` because it is frowned upon in some style guides and I, frankly, don't find it easier to read than an honest `for` loop - In particular, when errors need to be handled.